clients/admin-ui/src/features/config-wizard/AuthenticateAwsForm.tsx

```diff
@@ -110,101 +128,193 @@ const AuthenticateAwsForm = () => {
-                tooltip="Specify the default region in which your infrastructure is located. This is necessary for successful scanning."
-                options={AWS_REGION_OPTIONS}
-              />
-            </Stack>
-            <HStack>
-              <Button variant="outline" onClick={handleCancel}>
-                Cancel
+    <>
+      <AlertDialog
+        isOpen={isOpen}
+        leastDestructiveRef={cancelRef}
+        onClose={onClose}
+      >
+        <AlertDialogOverlay>
+          <AlertDialogContent alignItems="center" textAlign="center">
+            <WarningIcon marginTop={3} />
+            <AlertDialogHeader fontSize="lg" fontWeight="bold">
+              Cancel Scan!
+            </AlertDialogHeader>
+
+            <AlertDialogBody pt={0}>
+              <Text color="gray.500" mb={3}>
+                Warning, you are about to cancel the scan!
+              </Text>
+              <Text color="gray.500" mb={3}>
+                If you cancel scanning, the scanner will stop and no systems
+                will be returned.
+              </Text>
+              <Text color="gray.500" mb={3}>
+                Are you sure you want to cancel?
+              </Text>
+            </AlertDialogBody>
+
+            <AlertDialogFooter>
+              <Button onClick={() => handleCancel()} variant="outline">
+                Yes, Cancel
               </Button>
               <Button
-                type="submit"
-                variant="primary"
-                isDisabled={!dirty || !isValid}
-                isLoading={isLoading}
-                data-testid="submit-btn"
+                colorScheme="primary"
+                ref={cancelRef}
+                onClick={onClose}
+                ml={3}
               >
-                Save and Continue
+                No, Continue Scanning
               </Button>
-            </HStack>
-          </Stack>
-        </Form>
-      )}
-    </Formik>
+            </AlertDialogFooter>
+          </AlertDialogContent>
+        </AlertDialogOverlay>
+      </AlertDialog>
+      <Formik
+        initialValues={initialValues}
+        validationSchema={ValidationSchema}
+        onSubmit={handleSubmit}
+      >
+        {({ isValid, isSubmitting, dirty }) => (
+          <Form data-testid="authenticate-aws-form">
+            <Stack spacing={10}>
+              {isSubmitting ? (
+                <Text
+                  alignItems="center"
+                  as="b"
+                  color="gray.900"
+                  display="flex"
+                  fontSize="xl"
+                >
+                  System scanning in progress{" "}
+                  <CloseButton
+                    data-testid="close-scan-in-progress"
+                    display="inline-block"
+                    onClick={onOpen}
+                  />
+                </Text>
+              ) : null}
+              {!systemResultsNumber && isSubmitting ? (
```

Comment:
Using boolean checks with a number value is often a source of bugs/confusion. It is possible, if not likely, that there would be zero systems, and that would probably require different logic than the "results haven't come back yet" state. In fact, I might use the mutation's `isSuccess` check whether the request finished, and let this value just be a number.

Reply:
That's fair!